From ce87b765d501775a39c267492ee6411351f796b9 Mon Sep 17 00:00:00 2001
From: Reilly Brogan <reilly@reillybrogan.com>
Date: Thu, 23 Nov 2023 20:36:40 -0600
Subject: [PATCH] Desktop integration fixes

Changes:
- Match the app ID from Flathub
- Make sure we still match the ResourceClass when launched in x11
- Don't install the png icon, the svg is good enough

---
 CMakeLists.txt           | 5 ++---
 Data/dolphin-emu.desktop | 1 +
 2 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index b1eede22aa..73b5a51f24 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -836,10 +836,9 @@ if(CMAKE_SYSTEM_NAME MATCHES "Linux|FreeBSD|OpenBSD")
   # Install the application icon and menu item
   install(FILES Data/dolphin-emu.svg
           DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/scalable/apps)
-  install(FILES Data/dolphin-emu.png
-          DESTINATION ${CMAKE_INSTALL_PREFIX}/share/icons/hicolor/256x256/apps)
   install(FILES Data/dolphin-emu.desktop
-          DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications)
+          DESTINATION ${CMAKE_INSTALL_PREFIX}/share/applications
+          RENAME org.DolphinEmu.dolphin-emu.desktop)
   # Install manpages
   install(FILES Data/dolphin-emu.6
           DESTINATION ${CMAKE_INSTALL_MANDIR}/man6)
diff --git a/Data/dolphin-emu.desktop b/Data/dolphin-emu.desktop
index a998ce4370..cb7fef0ee5 100644
--- a/Data/dolphin-emu.desktop
+++ b/Data/dolphin-emu.desktop
@@ -4,6 +4,7 @@ Icon=dolphin-emu
 Exec=env QT_QPA_PLATFORM=xcb dolphin-emu
 Terminal=false
 Type=Application
+StartupWMClass=dolphin-emu
 Categories=Game;Emulator;
 Name=Dolphin Emulator
 GenericName=Wii/GameCube Emulator
-- 
2.43.0

